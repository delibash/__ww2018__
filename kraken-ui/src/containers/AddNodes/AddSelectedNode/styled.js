import styled from 'styled-components';

const StyledAddSelectedNode = styled.section`
  margin: 1rem 0 1.8rem 0;
`;

const StyledInputSection = styled.section`
  display: flex;
  & > input {
    flex: 1;
    margin-right: 1.8rem;
  }
`;

export {
  StyledAddSelectedNode,
  StyledInputSection
};