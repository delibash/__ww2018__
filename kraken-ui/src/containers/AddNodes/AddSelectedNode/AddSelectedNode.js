import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button, Input } from './../../../components/';
import { StyledInputSection, StyledAddSelectedNode } from './styled';

// TODO: ERROR BOUNDARY, https://reactjs.org/docs/error-boundaries.html

class AddSelectedNode extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      nodeName: '',
      selectedTitle: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleChange (e) {
    this.setState({
      ...this.state,
      selectedTitle: this.props.collection.selected,
      nodeName: e.target.value
    });
  }

  handleClick () {
    this.props.addSelectedNode(this.state);
    this.setState({
      ...this.state,
      nodeName: ''
    });
  }

  render () {
    const { collection, ui } = this.props;
    const { nodeName } = this.state;

    return (
      <>
        <StyledAddSelectedNode>
          <StyledInputSection>
            <Input
              placeholder={`Enter New ${collection && collection.selected}`}
              type="text"
              value={nodeName}
              onChange={this.handleChange}
            />
            <Button
              disabled={nodeName.length < 1 && true}
              onClick={this.handleClick}
            >
              {ui && !ui.nodeAdded ? 'Add Node' : `Add Another ${collection && collection.selected}`}
            </Button>
          </StyledInputSection>
        </StyledAddSelectedNode>
      </>
    );
  }
}

AddSelectedNode.propTypes = {
  collection: PropTypes.object,
  ui: PropTypes.object,
  addSelectedNode: PropTypes.func,
};

export default AddSelectedNode;
