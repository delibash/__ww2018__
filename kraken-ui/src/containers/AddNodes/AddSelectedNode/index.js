import AddSelectedNode from './AddSelectedNode';

export default AddSelectedNode;