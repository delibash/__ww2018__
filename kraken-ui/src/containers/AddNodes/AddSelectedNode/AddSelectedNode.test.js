import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import AddSelectedNode from './AddSelectedNode';

test('Renders AddSelectedNode', () => {
  const tree = renderer.create(
    <AddSelectedNode />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});