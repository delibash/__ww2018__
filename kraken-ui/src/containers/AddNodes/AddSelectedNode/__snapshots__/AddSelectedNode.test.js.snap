// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders AddSelectedNode 1`] = `
.c3 {
  padding: .8rem 1.8rem;
  text-transform: capitalize;
  font-weight: 300;
  font-size: 1.3rem;
  border: none;
  border-radius: .2rem;
  background-color: rgba(234,236,242,1);
  color: rgba(0,0,0,1);
  cursor: default;
}

.c3:hover {
  background-color: rgba(234,236,242,1);
  color: rgba(0,0,0,1);
}

.c2 {
  min-height: 3.8rem;
  padding: .2rem .8rem;
  border: 0;
  border-radius: .3rem;
  box-shadow: inset 0 0 0 .1rem rgba(0,0,0,.25);
  font-size: 1.3rem;
  width: 100%;
}

.c0 {
  margin: 1rem 0 1.8rem 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > input {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 1.8rem;
}

<section
  className="c0"
>
  <section
    className="c1"
  >
    <input
      className="c2"
      onChange={[Function]}
      placeholder="Enter New undefined"
      type="text"
      value=""
    />
    <button
      className="c3"
      disabled={true}
      onClick={[Function]}
    >
      Add Another undefined
    </button>
  </section>
</section>
`;
